mongodb/vibe/db/mongo/connection.d

```diff
@@ -306,26 +308,50 @@ final class MongoConnection {
 	{
 		scope(failure) disconnect();
 		foreach (d; documents) if (d["_id"].isNull()) d["_id"] = Bson(BsonObjectID.generate());
-		send(OpCode.Insert, -1, cast(int)flags, collection_name, documents);
+		string collection = collection_name.canFind(".") ? collection_name.split(".")[1] : collection_name;
+		Bson operation = Bson([
+			"insert": Bson(collection),
+			"$db": Bson(m_settings.database == string.init ? "admin" : m_settings.database),
+			"documents": Bson(documents)
+		]);
+		auto id = send(OpCode.Msg, -1, cast(int)0, cast(ubyte)0, operation);
```

Comment:
This uses `0` instead of the `flags` argument that was used for `OpCode.insert` - I'm not sure how the semantics translate to `OpCode.Msg`, though, probably they have to be added as fields to `operation`?